.github/workflows/gochecks.yaml

```diff
@@ -84,3 +84,8 @@ jobs:
               git status --short
               exit 1
             fi
+
+      - name: Run tests
+        run: |
+          go run github.com/onsi/ginkgo/v2/ginkgo -v -p -randomize-all \
+            ./cmdfactory/...
```

Comment:
Should this be part of the same commit? Also should this not be part of a test pipeline instead of a linter pipeline? See https://github.com/unikraft/kraftkit/issues/432

Reply:
My confusion results from the pipeline being named `golangci-lint`. This can be resolved just with renaming it to `gochecks`.

Reply:
For now, I'd remove this line (though it proves helpful for demonstrating how to test the code) and a separate PR can be opened to add this step "Run tests" and rename the pipeline.

Reply:
Done, see last commit.

Reply:
Ignore my last comment! :) Your force push was faster!

Reply:
I can also merge into "gochecks" and have 2 workflows inside. I'd prefer that a little actually :)

Reply:
I'd also move e2e into gotests as a parallel job along with units. Also, is this the right PR for this?

Reply:
Thoughts on merging go checks + go tests (unit + e2e) into a single pipeline? Or keep gochecks and gotests?

Reply:
You tell me, but I think it's critical that we at least run the tests included in this PR.

Reply:
I'd then do gochecks + gotests(unit+e2e) pipeline as a separate PR which is merged first and this PR is rebased.

Reply:
https://github.com/unikraft/kraftkit/issues/432

Reply:
Done #433
Will rebase after the merge.